lxd/storage/drivers/utils.go

```diff
@@ -732,3 +733,18 @@ func ShiftZFSSkipper(dir string, absPath string, fi os.FileInfo) bool {
 
 	return false
 }
+
+// blockDevSizeBytes returns the size of a block device.
+func blockDevSizeBytes(blockDevPath string) (int64, error) {
+	output, err := shared.RunCommand("blockdev", "--getsize64", blockDevPath)
+	if err != nil {
+		return -1, err
+	}
```

Comment:
This can be easily done with:

```go
        // Attempt to open the device path
        f, err := os.Open(devicePath)
        if err == nil {
                defer f.Close()
                fd := int(f.Fd())

                // Retrieve the block size
                res, err := unix.IoctlGetInt(fd, unix.BLKPBSZGET)
                if err != nil {
                        return err
                }

                disk.BlockSize = uint64(res)
        }
```
(taken from what we have in the resources API package)

Reply:
@stgraber thanks, but I don't think this is the same thing, as that ioctl is returning the block size, not the size of the block device.

Anyway, thanks for the tip, as that lead me to the `BLKGETSIZE64` ioctl, which is what I need here.

```
res, err := unix.IoctlGetInt(fd, unix.BLKGETSIZE64)
```